### ISSUE TITLE
Saturn and Saturn_lockfree dependencies

### ISSUE THREAD

**Issue by Sudha247** [ocaml-multicore/saturn#84]
I understand that `Saturn_lockfree` is intended to be an independent package people can use out of the box without relying on `Saturn`. This was not possible when I tried to do so in domainslib. The utop log shows it -

```ocaml
─( 17:18:04 )─< command 0 >──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────{ counter: 0 }─
utop # #require "saturn_lockfree";;
─( 17:18:05 )─< command 1 >──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────{ counter: 0 }─
utop # module Ws_deque = Saturn_lockfree.Work_stealing_deque.M;;
Error: Unbound module Saturn_lockfree
─( 17:19:36 )─< command 2 >──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────{ counter: 0 }─
utop # #require "saturn";;
─( 17:19:45 )─< command 3 >──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────{ counter: 0 }─
utop # module Ws_deque = Saturn.Work_stealing_deque.M;;
module Ws_deque = Saturn.Work_stealing_deque.M
─( 17:19:52 )─< command 4 >──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────{ counter: 0 }─
utop # 

```

Perhaps some dune magic needs to be done to recognise `Saturn_lockfree` as a module.

cc @lyrm

**Comment by lyrm** [comment]
HI! Actually, right now, to get the lockfree data structure, you need to install `saturn_lockfree` but the module itself can be called we `Lockfree` (like `Lockfree.Queue` for example). I will make the PR to correct that.

**Comment by Sudha247** [comment]
Thanks, I see. The module being called either `Lockfree` or `Saturn_lockfree` is fine. Since the package is called `Saturn_lockfree`, I'm inclined to call the module with the same name. We should make it clear in the docs.

**Comment by Sudha247** [comment]
Fixed in #85.
